pkg/remote/codec/protobuf/protobuf.go

```diff
@@ -79,6 +79,26 @@ func (c protobufCodec) Marshal(ctx context.Context, message remote.Message, out
 		return perrors.NewProtocolErrorWithMsg(fmt.Sprintf("protobuf marshal, write seqID failed: %s", err.Error()))
 	}
 
+	// If Using Generics
+	// if data is a MessageWriterWithContext
+	// Do msg.WritePb(ctx context.Context, out remote.ByteBuffer)
+	genmsg, ok := data.(MessageWriterWithContext)
```

Comment:
Is it possible to be placed within the `if !ok` block? Thus we introduce less performance impact to main path (a kitex client are more likely to be a normal client instead of a generic client)

Reply:
Yup, I shifted it to be within the `if !ok` block!